tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -56,6 +56,30 @@ TTreeProcessorMT::TTreeProcessorMT(TTree &tree) : treeView(tree) {}
 /// \param[in] entries List of entry numbers to process.
 TTreeProcessorMT::TTreeProcessorMT(TTree &tree, TEntryList &entries) : treeView(tree, entries) {}
 
+////////////////////////////////////////////////////////////////////////
+/// Divide input data in clusters, i.e. the workloads to distribute to tasks
+std::vector<ROOT::Internal::TreeViewCluster> TTreeProcessorMT::MakeClusters()
+{
+   std::vector<ROOT::Internal::TreeViewCluster> clusters;
+   const auto &fileNames = treeView->GetFileNames();
+   const auto nFileNames = fileNames.size();
+   const auto &treeName = treeView->GetTreeName();
+   for (auto i = 0u; i < nFileNames; ++i) { // we need the indeces as we pass them to TreeViewCluster's ctor below
```

Comment:
I keep getting it wrong don't I? :sweat_smile: 

The index of the filename is one of three data members of TreeViewCluster, so of course the ctor needs it. A TreeViewCluster needs the filename index otherwise there's no way to tell which file the cluster pertains to.